src/main/scala/elevate/rise/rules/lowering.scala

```diff
@@ -74,4 +115,92 @@ object lowering {
     }
     override def toString = s"slideSeq($rot, $write_dt1)"
   }
+
+  // Lowerings used in PLDI submission
+
+  // adds copy after every generate
```

Comment:
Can we have a discussion on this "copy" terminology here?
For me the important thing is that we want to write it / to store it in memory.
On the functional side there is an identity, on the low-level side there is a memory write, why are we talking about a copy?
@bastian-koepcke @bastianhagedorn